### PR TITLE
[TRANSFORMATIONS] Add EliminateConvPaddingMaskGating pass to SDPAToPA

### PR DESCRIPTION
**Pull request by CuriousPanCake**
### Details:
 - For the LFM2 model exported using transformers 5.0 it was observed the presence of a padding subgraph built from the attention_mask input. For the execution in the Continuous Batching mode it is not needed as well as the attention mask input. During the existing SDPAToPA passes it is not removed, so requires an additional transformation for capturing and removing it. Add EliminateConvPaddingMaskGating pass for it.

### Tickets:
 - [CVS-186696](https://jira.devtools.intel.com/browse/CVS-186696)

### AI Assistance:
 - *AI assistance used: yes*

Signed-off-by: Andrii Staikov <andrii.staikov@intel.com>

### CODE REVIEW

File: src/common/transformations/tests/op_conversions/sdpa_to_paged_attention_test.cpp [hunk 1]
```diff
@@ -59,6 +59,7 @@
 #include "openvino/op/util/variable.hpp"
 #include "openvino/op/variadic_split.hpp"
 #include "openvino/pass/visualize_tree.hpp"
```
Comment (CuriousPanCake) [inline src/common/transformations/tests/op_conversions/sdpa_to_paged_attention_test.cpp]:
```suggestion
```

File: src/core/src/pass/sdpa_to_paged_attention.cpp [hunk 2]
```diff
@@ -79,6 +82,8 @@ bool ov::pass::SDPAToPagedAttention::run_on_model(const std::shared_ptr<ov::Mode
                     "No ScaledDotProductAttention operation observed in the graph, cannot perform "
                     "the SDPAToPagedAttention transformation.");
 
+    // ov::pass::Serialize("gemma_sdpa_to_paged_attention_before.xml", "gemma_sdpa_to_paged_attention_before.bin").run_on_model(model);
+
```
Comment (CuriousPanCake) [inline src/core/src/pass/sdpa_to_paged_attention.cpp]:
```suggestion
```

File: src/core/src/pass/sdpa_to_paged_attention.cpp [hunk 3]
```diff
@@ -197,7 +203,10 @@ bool ov::pass::SDPAToPagedAttention::run_on_model(const std::shared_ptr<ov::Mode
 
     model->add_results(m_results.items());
     model->add_parameters(m_params.items());
+    ov::pass::VisualizeTree("sdpa_to_paged_attention_after.svg").run_on_model(model);
+    std::cout << "before validate infer types" << std::endl;
```
Comment (CuriousPanCake) [inline src/core/src/pass/sdpa_to_paged_attention.cpp]:
```suggestion
```

File: src/core/src/pass/sdpa_to_paged_attention.cpp [hunk 4]
```diff
@@ -197,7 +203,10 @@ bool ov::pass::SDPAToPagedAttention::run_on_model(const std::shared_ptr<ov::Mode
...
     model->validate_nodes_and_infer_types();
+    std::cout << "after validate infer types" << std::endl;
```
Comment (CuriousPanCake) [inline src/core/src/pass/sdpa_to_paged_attention.cpp]:
```suggestion
```

File: src/core/src/pass/sdpa_to_paged_attention.cpp [hunk 5]
```diff
@@ -26,6 +27,8 @@
 #include "transformations/utils/print_model.hpp"
 #include "transformations/utils/utils.hpp"
 
+#include "openvino/pass/visualize_tree.hpp"
+
```
Comment (CuriousPanCake) [inline src/core/src/pass/sdpa_to_paged_attention.cpp]:
```suggestion
```

File: src/common/transformations/src/transformations/paged_attention/eliminate_conv_padding_mask_gating.cpp [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+
+EliminateConvPaddingMaskGating::EliminateConvPaddingMaskGating() {
+    MATCHER_SCOPE(EliminateConvPaddingMaskGating);
+
+    // Pattern: attention_mask -> Slice -> Unsqueeze -> Convert -> Multiply -> Add -> Multiply(H, mask_expr)
+    auto attn_mask = wrap_type<v0::Parameter>([](const ov::Output<ov::Node>& output) {
+        return output.get_node_shared_ptr()->get_friendly_name() == "attention_mask";
+    });
+    auto slice = wrap_type<v8::Slice>({attn_mask, any_input(), any_input(), any_input(), any_input()});
+    auto unsqueeze = wrap_type<v0::Unsqueeze>({slice, any_input()});
```
Comment (mryzhov) [inline src/common/transformations/src/transformations/paged_attention/eliminate_conv_padding_mask_gating.cpp]:
```suggestion
    auto unsqueeze = pattern::optional<v0::Unsqueeze>({slice, any_input()});
```

Comment (CuriousPanCake) [reply src/common/transformations/src/transformations/paged_attention/eliminate_conv_padding_mask_gating.cpp]: I don't think this is necessary. Concerning the complexity of this graph, the Unsqueeze most probably is going to be here.

Comment (mryzhov) [reply src/common/transformations/src/transformations/paged_attention/eliminate_conv_padding_mask_gating.cpp]: then explain please the importance of this layer in the pattern

File: src/common/transformations/src/transformations/paged_attention/eliminate_conv_padding_mask_gating.cpp [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+    auto convert = pattern::optional<v0::Convert>({unsqueeze});
+    auto mul_mask = wrap_type<v1::Multiply>({convert, any_input()});
+    auto add = wrap_type<v1::Add>({mul_mask, any_input()});
+    auto hidden_states = any_input();
+    auto mul_gate = wrap_type<v1::Multiply>({hidden_states, add});
```
[the remark this answers is not in the document]
Comment (CuriousPanCake) [reply src/common/transformations/src/transformations/paged_attention/eliminate_conv_padding_mask_gating.cpp]: Fixed to get_names()

File: src/common/transformations/include/transformations/paged_attention/eliminate_conv_padding_mask_gating.hpp [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+ * @brief Eliminates the conv padding mask gating subgraph introduced by transformers 5.0+.
+ *
+ * Transformers 5.0 added `apply_mask_to_padding_states` which multiplies hidden_states
+ * by attention_mask before conv layers. First observed in the LFM2 model.
+ * In PagedAttention mode, sequences are packed contiguously without padding, so this gating is
+ * always an identity (mask is all-1s).
+ *
+ * Matches the pattern:
+ * attention_mask -> Slice -> Unsqueeze -> Convert -> Multiply -> Add -> Multiply(H, mask_expr)
+ * and replaces the final Multiply with its hidden_states input directly.
```
[the remark this answers is not in the document]
Comment (CuriousPanCake) [reply src/common/transformations/include/transformations/paged_attention/eliminate_conv_padding_mask_gating.hpp]: Fixed

**Review by mryzhov — APPROVED** [review]
